File: people_database/people_database.py
from person import Person

class PeopleDatabase:
    def __init__(self, person_data):
        self.people = []
        if type(person_data) == list:
            self.people = person_data
        else:
            self.people.append(person_data)

    def add(self, first, last, em, st):
        person = Person(first, last, em, st)
        self.people.append(person)

    def people(self):
        return self.people


    def return_all_emails(self):
        new_list = []
        for i in self.people:
            new_list.append(i.email)
        return new_list

    def return_all_by_state(self, state):
        new_list = []
        for i in self.people:
            if i.state == state:
                new_list.append(i.info())

        return new_list


    def remove_by_email(self, email):
        self.people = [x for x in self.people if x.email != email]

    def count_all_from_state(self, state):
        new_list = [x for x in self.people if x.state == state]
        return len(new_list)

File: people_database/people_database.rb
require_relative "person"

class PeopleDatabase
  attr_reader :people

  def initialize(people)
    @people = []
    @people << people

  end

  def add(person_data)
    people << Person.new(person_data)
  end

  def remove_by_email(email)
    people.delete_if do |person|
      person.email == email
    end

  end

  def return_all_by_state(state)
    people.select do |person|
      person.state == state
    end
  end

  def return_all_emails
    people.map do |person|
      person.email
    end.join(", ")
  end

  def count_all_from_state(state)
    return_all_by_state(state).count
  end

end
